src/citrine/_session.py

```diff
@@ -35,7 +37,7 @@ def __init__(self,
                  port: Optional[str] = None):
         super().__init__()
         self.scheme: str = scheme
-        self.authority = ':'.join([host or '', port or ''])
+        self.authority = ':'.join([host] if host else [] + [port] if port else [])
```

Comment:
This might actually be better written as 
```suggestion
        self.authority = ':'.join([host] + [port] if port else [])
```
since the host cannot be `None` if the signature is being respected.

Reply:
But there's nothing stopping a user from passing `host=None`. Python just doesn't demand that kind of respect :p

Is the point of this change that, when `port` isn't specified, the url should be `https://host/` instead of `https://host:/`?

Reply:
That is an issue that both versions of the line address.  If we want to be strictly compliant with the spec, we should have `https:FAQ.html` as a relative path when no host or path is specified -- the unsplit should handle that, though we'll get a bad result if port is specified and host is None.

I'm going with this proposed change should be axed.